### ISSUE TITLE
Uncaught Error: socket hang up

### ISSUE THREAD

**Issue by tungdt-90** [atom/atom#5048]
[Enter steps to reproduce below:]

1. ...
2. ...

**Atom Version**: 0.169.0
**System**: linux 3.13.0-43-generic
**Thrown From**: Atom Core

### Stack Trace

Uncaught Error: socket hang up

```
At stream.js:94

Error: socket hang up
  at createHangUpError (_http_client.js:214:15)
  at TLSSocket.socketOnEnd (_http_client.js:292:23)
  at TLSSocket.emit (events.js:129:20)
  at _stream_readable.js:907:16
  at process._tickCallback (node.js:386:11)

```

### Commands

```
     -0:19.8 command-palette:toggle (atom-text-editor.editor)
  4x -0:12.8 core:backspace (atom-text-editor.editor.mini)
     -0:06.2 core:move-up (atom-text-editor.editor.mini)
     -0:05.6 core:confirm (atom-text-editor.editor.mini)
     -0:05.6 settings-view:install-packages-and-themes (atom-text-editor.editor)
     -0:00.0 core:confirm (atom-text-editor.editor.mini)
```

### Config

```json
{
  "core": {
    "disabledPackages": [
      "metrics"
    ],
    "themes": [
      "atom-dark-ui",
      "monokai"
    ]
  },
  "editor": {
    "fontSize": 14,
    "showIndentGuide": true,
    "confirmCheckoutHeadRevision": false,
    "invisibles": {}
  }
}
```

### Installed Packages

```coffee
# User
auto-indent, v0.1.0
monokai, v0.9.0
project-manager, v1.14.1

# Dev
No dev packages
```

**Comment by 50Wliu** [comment]
@tungdt-90 What were you doing when this happened?  It looks like you were trying to install a package-are you behind a firewall by any chance?

**Comment by tungdt-90** [comment]
Hi ! Yeah, i got this problem when i'm trying to install package "Minimap"
for Atom. Actually, in these 2 weeks, i cannot install any package by
setting in Atom. Instead of that, i have to use command "apm install..."
and everything go normal with the command.

**Comment by kevinsawicki** [comment]
I believe this was fixed by https://github.com/atom/settings-view/commit/3fe0db5dc1c7db9f936b506a126d689057a24e0d which was included in the latest release, 0.172
